permission_handler_web/lib/web_delegate.dart

```diff
@@ -0,0 +1,223 @@
+import 'dart:html' as html;
+import 'dart:async';
+
+import 'package:permission_handler_platform_interface/permission_handler_platform_interface.dart';
+
+/// The delegate class for WebPermissionHandler.
+/// Used for dependency injection of html.MediaDevices and html.Permissions objects
+class WebDelegate {
+  /// Constructs a WebDelegate.
+  WebDelegate(this.devices, this.geolocation, this.htmlPermissions);
+
+  /// The html media devices object used to request camera and microphone permissions.
+  html.MediaDevices? devices;
+
+  /// The html geolocation object used to request location permission.
+  html.Geolocation? geolocation;
+
+  /// The html permissions object used to check permission status.
+  html.Permissions? htmlPermissions;
```

Comment:
These can should all be `final` and private fields, they matter internally only and should not be accessed from outside this class.

 ```suggestion
  /// The html media devices object used to request camera and microphone permissions.
  final html.MediaDevices? _devices;

  /// The html geolocation object used to request location permission.
  final html.Geolocation? _geolocation;

  /// The html permissions object used to check permission status.
  final html.Permissions? _htmlPermissions;
```

This means you'd have to update the constructor as well:

```dart
  WebDelegate(
      html.MediaDevices? devices, 
      html.Geolocation? geolocation, 
      html.Permissions? permissions,
  ) : _devices = devices,
       _geolocation = geolocation,
       _htmlPermissions = permissions;
```